torch/linalg/__init__.py

```diff
@@ -1419,69 +1481,71 @@
+    >>> torch.linalg.pinv(A, rcond)
+
+.. _defined algebraically:
+    https://en.wikipedia.org/wiki/Moore%E2%80%93Penrose_inverse#Existence_and_uniqueness
+.. _through the SVD:
+    https://en.wikipedia.org/wiki/Moore%E2%80%93Penrose_inverse#Singular_value_decomposition_(SVD)
 """)
 
 solve = _add_docstr(_linalg.linalg_solve, r"""
-linalg.solve(input, other, *, out=None) -> Tensor
-
-Computes the solution of a square system  of linear equations with unique solution. Supports batched inputs.
-
-This method accepts both matrices and vectors as the right-hand side of the linear equation.
-We will denote by `*` zero or more batch dimensions.
+linalg.solve(A, B, *, out=None) -> Tensor
 
-- If :attr:`input`, :attr:`other` have shapes `(*, n, n)`, `(*, n)`, for every matrix :math:`A`
-  in :attr:`input` and vector :math:`b` in :attr:`other`, this function returns a vector :math:`x` such that
+Computes the solution of a square system of linear equations with unique solution.
 
-  .. math:: Ax = b
+Letting :math:`\mathbb{K}` be :math:`\mathbb{R}` or :math:`\mathbb{C}`,
+this function computes a solution :math:`X \in \mathbb{K}^{n \times k}` of the linear system associated to
+:math:`A \in \mathbb{K}^{n \times n}, B \in \mathbb{K}^{m \times k}`, which is defined as
 
-  The returned tensor have shape `(*, n)`.
-- If :attr:`input`, :attr:`other` have shapes `(*, n, n)`, `(*, n, k)`, for every pair of matrices
-  :math:`A` in :attr:`input` and :math:`B` in :attr:`other`, this function returns a matrix :math:`X` such that
+.. math:: AX = B
 
-  .. math:: AX = B
+Supports inputs of float, double, cfloat and cdouble dtypes.
+Also supports batched inputs, and, if the input is batched, the output is batched with the same dimensions.
 
-  The returned tensor have shape `(*, n, k)`.
-- If none of the above apply and :attr:`input`, :attr:`other` have shapes `(*, n, n)`, `(n,)` (resp. `(n, k)`),
-  the vector (resp. matrix) :attr:`other` is broadcasted to have shape `(*, n)` (resp. `(*, n, k)`).
-  This function then  returns the solution of the resulting batch of systems of linear equations.
+Denote by `*` zero or more batch dimensions.
 
-Supports :attr:`input` of float, double, cfloat and cdouble dtypes.
-:attr:`other` should have the same dtype as :attr:`input`.
-
-.. note:: The solution to these problems can be computed in terms of the inverse of `A`
-          as `A.inv() @ b` and `A.inv() @ B` respectively.
-          This function can be seen as faster and more stable way of computing these quantities.
+- If :attr:`A` has shape `(*, n, n)` and :attr:`B` has shape `(*, n)` (a batch of vectors) or shape
+  `(*, n, k)` (a batch of matrices or "multiple right-hand sides"), this function returns `X` of shape
+  `(*, n)` or `(*, n, k)` respectively.
+- Otherwise, if :attr:`A` has shape `(*, n, n)` and  :attr:`B` has shape `(n,)`  or `(n, k)`, :attr:`B`
+  is broadcasted to have shape `(*, n)` or `(*, n, k)` respectively and returns the solution of the resulting
```

Comment:
the "returns" here refers to B, but it should refer to the function

this section needs a rethink to clarify these cases

Reply:
I changed it a bit, but yeah, happy to take suggestions on how to restructure this section.